src/objects/move_data.rs

```diff
@@ -69,15 +74,30 @@ impl MoveController for UserController {
             dir.x = -1.0;
         }
 
+        // Speed as f(inertia)
         let target_velocity = if dir.norm() == 0.0 {
             dir
         } else {
             dir / dir.norm() * self.speed
         };
 
         let diff_velocity = target_velocity - move_data.velocity;
-
         move_data.velocity += diff_velocity * (1.0 - self.inertia);
+
+        /* Mouse input to angle (aka. What the fuck am I doing???)
+        let m = mouse::position(ctx); // Type Point2
+        let mouse_pos = Vector2::new(m.x, m.y); // Transformed to Vector2 to be read by Matrix
+        let current_angle = Matrix::angle(&mouse_pos, &move_data.position); */
+
+        let m = mouse::position(ctx);
+        move_data.angle = ((m.y - move_data.position.y).atan2(move_data.position.x - m.x)) * (consts::PI / 2.0) as f32;
```

Comment:
Note that atan2 already returns radians, so no need to multiply by Pi/2. Also, notice on one argument you're doing `m.y - position.y` and on the other you're doing `position.x - m.x`

Reply:
I understand, but I've noticed some issues I'd like to discuss when possible.
1. m.y - position.y is like that because, as pixels on the screen, y downwards is positive while for getting the angle I would like y upwards to be positive, so I would need to change the signs and this changes the order (otherwise it would be -position.y -(-m.y)).
2. I just removed pi/2 but there's still not much of a change. I also found recently that, according to Wikipedia, "atan2(y,x) returns the angle θ between the ray to the point (x,y) and the positive x-axis, confined to (-π, π]." -- Positive x-axis (only?) confined to -pi and +pi? Note also that the graph of atan2 seems to not give the result we want. https://en.wikipedia.org/wiki/Atan2#/media/File:Arctangent2.svg
3. Finally, I was checking what data move_data.position was giving me and I noticed that the coordinate system (where 0,0 is on screen) does not correspond to the coordinate system of pixels that I get with the mouse position function. This is why I tried to represent y as - (position.y - m.y) in atan2 but this didn't work :/

Reply:
Ah, I know now what the issue is. Basically, the position coordinates are in "world coordinates" (aka relative to the center of the map vs actual screen positions). I'll send a PR that will let you convert screen coordinates over to world coordinates. That way you just need to plug in the mouse position to that function and you'll get something in the same coordinate system as the player

Reply:
Alright, #23 adds what you need